[NNC] Fix loopnest.cache_accesses for reduce ops (fixed #59002)

huiguoo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59136 [NNC] Fix loopnest.cache_accesses for reduce ops (fixed #59002)**

Differential Revision: [D28768598](https://our.internmc.facebook.com/intern/diff/D28768598)

huiguoo committed:
[NNC] Fix loopnest.cache_accesses for reduce ops (fixed #59002)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28768598).

ZolotukhinM commented on test/cpp/tensorexpr/test_reductions.cpp:
```diff
@@ -1331,34 +1331,138 @@ TEST(Reductions, ReductionCacheAccessesOuter) {
+#CHECK:   }
+#CHECK: }
+#CHECK: for (int i
+#CHECK:   sum[i] = d_local[i]
```
Just for my understanding: this is the case that was incorrect previously, right? Were we generating a reduceOp here?

huiguoo replied:
Yes, that's the case we hit before: the `cache_accesses` is performed on the operator axis and we still got a reduceOp which we shouldn't.

ZolotukhinM reviewed (approved):
Looks good to me!

huiguoo committed:
Update on "[NNC] Fix loopnest.cache_accesses for reduce ops (fixed #59002)"


Differential Revision: [D28768598](https://our.internmc.facebook.com/intern/diff/D28768598)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28768598).
